pydicom/dataelem.py

```diff
@@ -169,6 +173,134 @@ def __init__(self,
+                            set(val.keys()) & set(jsonrep.JSON_VALUE_KEYS)
+                        )
+                        if len(unique_value_keys) == 0:
+                            logger.debug(
+                                'data element has neither key "{}".'.format(
+                                    '" nor "'.join(supported_keys)
+                                )
+                            )
+                            elem = DataElement(tag=tag, value='', VR=vr)
+                        else:
+                            value_key = unique_value_keys[0]
+                            elem = cls.from_json(
+                                dataset_class, key, val['vr'],
+                                val[value_key], value_key
+                            )
+                        ds.add(elem)
+                elem_value.append(ds)
+        elif vr == 'PN':
+            # Special case, see DICOM Part 18 Annex F2.2
+            elem_value = []
+            for v in value:
+                if not isinstance(v, dict):
+                    # Some DICOMweb services get this wrong, so we
+                    # workaround the the issue and warn the user
+                    # rather than raising an error.
+                    logger.error(
+                        'value of data element "{}" with VR Person Name (PN) '
+                        'is not formatted correctly'.format(tag)
+                    )
+                    elem_value.append(v)
+                else:
+                    # import pdb;pdb.set_trace()
+                    elem_value.extend(list(v.values()))
+            if vm == '1':
+                try:
+                    elem_value = elem_value[0]
+                except IndexError:
+                    elem_value = ''
+        else:
+            if vm == '1':
+                if value_key == 'InlineBinary':
+                    elem_value = base64.b64decode(value)
+                elif value_key == 'BulkDataURI':
+                    if bulk_data_uri_handler is None:
+                        logger.warning(
+                            'no bulk data URI handler provided for retrieval '
+                            'of value of data element "{}"'.format(tag)
+                        )
+                        elem_value = ''
+                    else:
+                        elem_value = bulk_data_uri_handler(value)
+                else:
+                    if value:
+                        elem_value = value[0]
+                    else:
+                        elem_value = value
+            else:
+                elem_value = value
+        if not value:
+            logger.warning('missing value for data element "{}"'.format(tag))
+            elem_value = ''
+
+        elem_value = jsonrep._convert_to_python_number(elem_value, vr)
+
+        try:
+            if compat.in_py2 and vr == "PN":
+
+                elem_value = PersonNameUnicode(elem_value, None)
```

Comment:
Shouldn't we use utf-8 encoding here? I think this is what json uses by default.

Reply:
Sounds good.  I'll add that.